docs/source/dev/performance.rst

```diff
@@ -221,3 +223,170 @@ areas:
 .. The optimization reports provided by the Intel fortran compiler give a static
 .. analysis of code optimization. Specifically, the vectorization and openmp
 .. reports were analyzed to determine
+
+
+High-performance programming in Fortran
+---------------------------------------
+Being a compiled language designed for scientific and engineering
+applications, Fortran is well suited for producing very efficient
+code. However, the process of tuning code requires developers to
+understand the language as well as the tools available (compilers,
+performance libraries, etc) in order to generate the highest
+performance. This section identifies programming patterns to use
+Fortran and the Intel® Fortran compiler most effectively.
+
+Optimization report
+~~~~~~~~~~~~~~~~~~~
+Fortran compilers have multiple levels of optimization available
+from no optimization to extreme optimization tuned to a particular
+machine architecture and operating system combination. Timing tests
+alone are not a good indication for the compiler's ability to optimize
+a particular portion of code. Therefore, developers should generate
+optimization reports to get line-by-line reporting on optimizations
+such as vectorization, parallelization, memory and cache usage,
+threading, and others.
+
+https://software.intel.com/content/www/us/en/develop/articles/vectorization-and-optimization-reports.html
+
+- Link to flags for optimization report settings
+- Detail the process for generating the optrpt
```

Comment:
Are these bullets placeholders for future details?

Reply:
Yes, I've marked this pull request as a draft. There's still a bit of work to do.